### PR TITLE
fix(API): allow nullable value for job's due_date

### PR DESCRIPTION
**Pull request by hahmed-dev**
Purpose:
When creating a job from CLI and not providing `due_date` was adding a dummy due_date, which in fact was Go Lang adding a [zero value](https://go.dev/play/p/vTidOlmb9P) to an attribute `due_date` of Time type if no value is provided. 
This PR updates the schema to allow `nullable` values against `due_date` for `create`, `update` requests as well as when fetching the job.

Relevant Ticket: https://phrase.atlassian.net/browse/TSI-2297

### CODE REVIEW

File: openapi-generator/templates/java/libraries/okhttp-gson/build.gradle.mustache [hunk 1]
```diff
@@ -142,6 +142,7 @@ dependencies {
     implementation 'com.squareup.okhttp3:logging-interceptor:3.14.7'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.gsonfire:gson-fire:1.8.4'
+    implementation 'org.openapitools:jackson-databind-nullable:0.2.1'
```
Comment (hahmed-dev) [inline openapi-generator/templates/java/libraries/okhttp-gson/build.gradle.mustache:145]: The reason for adding this line was failure encountered in [Java tests](https://github.com/phrase/openapi/actions/runs/7755448581/job/21150828820) regarding package resolution.

File: clients/go/test/api_jobs_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			// assert request body
+			var requestBody map[string]interface{}
+			decoder := json.NewDecoder(r.Body)
+			if err := decoder.Decode(&requestBody); err != nil {
+				t.Fatal(err)
+			}
+
+			// Access the values from the JSON payload
+			assert.Equal(t, requestBody["name"], "createjobactionb1")
+			assert.Equal(t, requestBody["due_date"], nil)
```
Comment (jablan) [inline clients/go/test/api_jobs_test.go:36]: I presume the test fails when the change in `/path/jobs` is reverted?

**Review by jablan — APPROVED** [review]

**Comment by jablan** [comment]
minor comment: I wouldn't title this `fix(GO)`, rather `fix(API)` as it fixes the schema itself.

File: clients/go/test/api_jobs_test.go (hunk 2 above)
Comment (hahmed-dev) [reply clients/go/test/api_jobs_test.go:36]: Yes, let me add some screenshot to description to make it clear. :)

Comment (hahmed-dev) [reply clients/go/test/api_jobs_test.go:36]: @jablan when I remove nullable from `create.yaml`

Screenshot 2024-02-05 at 09 43 02

Comment (jablan) [reply clients/go/test/api_jobs_test.go:36]: thanks, I wasn't really doubting :)

**Pull request merged**